src/app/dynamic/enterprise/quotas/component.spec.ts

```diff
@@ -0,0 +1,267 @@
+//    explicitly provided under this license, the binary code.
+// 2. Any use of the software which exceeds the foregoing right, including,
+//    without limitation, its execution, compilation, copying, modification
+//    and distribution, is expressly prohibited.
+// 3. THE SOFTWARE IS PROVIDED “AS IS”, WITHOUT WARRANTY OF ANY KIND,
+//    EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+//    MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+//    IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY
+//    CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
+//    TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
+//    SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+//
+// END OF TERMS AND CONDITIONS
+
+import {NoopAnimationsModule} from '@angular/platform-browser/animations';
+import {SettingsMockService} from '@test/services/settings-mock';
+import {ComponentFixture, TestBed} from '@angular/core/testing';
+import {BrowserModule, By} from '@angular/platform-browser';
+import {UserMockService} from '@test/services/user-mock';
+import {SettingsService} from '@core/services/settings';
+import {MatTableModule} from '@angular/material/table';
+import {UserService} from '@core/services/user';
+import {SharedModule} from '@shared/module';
+import {QuotasComponent} from './component';
+import {lastValueFrom} from 'rxjs';
+import {map} from 'rxjs/operators';
+
+describe('QuotasComponent', () => {
+  let fixture: ComponentFixture<QuotasComponent>;
+  let component: QuotasComponent;
+
+  let settingService: SettingsService;
+  let userService: UserService;
+
+  const getQuotas = () =>
+    // @ts-ignore
+    lastValueFrom(settingService._getQuotas().pipe(map(quotas => quotas.filter(quota => !!quota))));
+
+  const setQuotas = async () => {
+    component.quotas = await getQuotas();
+    component.setDataSource();
+    fixture.detectChanges();
+  };
+
+  beforeEach(async () => {
+    await TestBed.configureTestingModule({
+      imports: [BrowserModule, NoopAnimationsModule, SharedModule, MatTableModule],
+      declarations: [QuotasComponent],
+      providers: [
+        {provide: SettingsService, useClass: SettingsMockService},
+        {provide: UserService, useClass: UserMockService},
+      ],
+    }).compileComponents();
+  });
+
+  beforeEach(async () => {
+    fixture = TestBed.createComponent(QuotasComponent);
+    component = fixture.componentInstance;
+
+    settingService = TestBed.inject(SettingsService);
+    userService = TestBed.inject(UserService);
+
+    fixture.detectChanges();
+  });
+
+  it('should initialize', async () => {
+    expect(component).toBeTruthy();
+  });
+
+  it('should display table when quotas are loaded', async () => {
+    component.isLoading = false;
+    component.quotas = await getQuotas();
+    fixture.detectChanges();
+
+    const element = fixture.debugElement.query(By.css('table'));
```

Comment:
Can we add explicit Id? Since requirement gets so updated what if we have multiple tables within DOM? 
- What if we have Tab functionality? If you think it's okay than IGNORE this comment i can accept is resolved.

Just my opinion
(By.css('#quotas-table'));


If you update it -> Update all placed below in TEST cases

Reply:
I think its fine for now, according to YAGNI principal, if in the future we have multiple tables there then I'll update the testing logic too